### ISSUE TITLE
NMAP using all RAM (over 64GB when scanning QNAP device)

### ISSUE THREAD

**Issue by RamboRogers** [nmap/nmap#2024]
Hello NMAP Team,

I'm running this scan, I've upgraded to the latest Nmap in Debian. 

`nmap -A -PS -PE -sS -vv --host-timeout 15m --script-timeout 2m --script (default or discovery or auth) and not (broadcast or brute or dos or exploit or external or fuzzer or intrusive) -oX scans/RANGE-192.168.1.1-192.168.1.254/current/192/168/1/4.xml 192.168.1.4`

```
root@ivreclient:/# nmap -v
Starting Nmap 7.70 ( https://nmap.org ) at 2020-04-24 21:11 UTC
Read data files from: /usr/local/bin/../share/nmap
WARNING: No targets were specified, so 0 hosts scanned.
Nmap done: 0 IP addresses (0 hosts up) scanned in 0.03 seconds
           Raw packets sent: 0 (0B) | Rcvd: 0 (0B)
root@ivreclient:/#
```

> 27356 27234 root     R    **33.3g**  50%   5  12% nmap -A -PS -PE -sS -vv --host-timeout 15m --script-timeout 2m --script (defau

How can I get better support data?

Thanks for the help.  Love and appreciate all of your hard work on this tool!

**Comment by dmiller-nmap** [comment]
Thanks for the report. I'm looking into some ways to implement warning output when NSE consumes lots of memory, but that will only help us get bug reports earlier. To solve the particular issue here, we need to try to narrow down which script is having problems:

* What ports are open on the target, and what services are they running?
* Does removing `or auth` or `or discovery` from your `--script` option prevent the excessive memory usage? Which one?

Using the answers to those questions, we should be able to narrow down which scripts are running. 

Of note, Nmap 7.80 introduced memory limits to the [`http` NSE library](https://nmap.org/nsedoc/lib/http.html) (#1571), so if it looks like a http-related script is the culprit, upgrading to Nmap 7.80 may solve the issue.

**Comment by RamboRogers** [comment]
Thank you Mr. Miller, I'm running the scan via IVRE [https://ivre.rocks/](https://ivre.rocks/).
I just ran the same scan on my MacBook using 7.8, didn't have the ram issue.  Must be fixed in 7.8.  I realized that this IVRE is using Debian stable.... which is super outdated.

**Comment by goarblt** [comment]
hello
https://www.evil.com/
